pkg/engine/engine.go

```diff
@@ -272,8 +272,16 @@ func (e *Engine) detectorWorker(ctx context.Context) {
 					ctx.Logger().Info("unknown decoder type", "type", reflect.TypeOf(decoder).String())
 					decoderType = detectorspb.DecoderType_UNKNOWN
 				}
+
+				original := chunk.Data
 				decoded := decoder.FromChunk(chunk)
-				if decoded == nil {
+
+				if decoded == nil ||
+					// check if the decoded data is similar "enough" to the original data. If it is, then we can skip scanning the decoded data as
+					// it's likely already picked up by the PLAIN decoder. See related issue: https://github.com/trufflesecurity/trufflehog/issues/1450
+					(decoded != nil &&
+						decoderType == detectorspb.DecoderType_BASE64 &&
+						bytes.Equal(original[0:common.MinInt(40, len(original)-1)], decoded.Data[0:common.MinInt(40, len(decoded.Data)-1)])) {
```

Comment:
very nit: I could see a small benefit of pulling this into a helper function. Then again there is also the comment so it's probably fine this way.

I was thinking maybe something like:
```
func areFirstNBytesEqual(a, b []byte, numBytes int) bool {
	limit := MinInt(numBytes, MinInt(len(a), len(b)))
	return bytes.Equal(a[:limit], b[:limit])
}
```

Reply:
Good call, that makes it more readable